src/Microsoft.AspNetCore.Blazor/Rendering/Renderer.cs

```diff
@@ -113,7 +113,28 @@ internal void InstantiateChildComponentOnFrame(ref RenderTreeFrame frame)
         internal void AssignEventHandlerId(ref RenderTreeFrame frame)
         {
             var id = ++_lastEventHandlerId;
-            _eventHandlersById.Add(id, (UIEventHandler)frame.AttributeValue);
+
+            // The attribute value might be a more specialized type like UIKeyboardEventHandler.
+            //
+            // In order to dispatch the event, we need a UIEventHandler, so we're going weakly
+            // typed here. The user will get a cast exception if they map the wrong type of
+            // delegate to the event.
+            if (frame.AttributeValue is UIEventHandler wrapper)
```

Comment:
I will go and read the code until I understand it properly, but at first glance I'm struggling to tell why it's sometimes a `UIEventHandler` and other times a `MulticastDelegate`. Might be useful to have a comment specifying that explicitly.

Reply:
Sure